infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientConfigListener.java

```diff
@@ -0,0 +1,218 @@
+            started.set(true);
+        } catch (StreamingException e) {
+            if (e.getExceptionCause().equals(StreamingException.ExceptionCause.LISTENER_SUBSCRIBED)) {
+                log.error("Stream listener already registered!");
+            } else {
+                log.error("Stream listener subscribe failed!", e);
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * A corfu update can/may have multiple updates belonging to different streams.
+     * This callback will return those updates as a list grouped by their Stream UUIDs.
+     * <p>
+     * Note: there is no order guarantee within the transaction boundaries.
+     *
+     * @param results is a map of stream UUID -> list of entries of this stream.
+     */
+    @Override
+    public void onNext(CorfuStreamEntries results) {
+        log.info("Client config listener onNext: {}", results.getEntries().size());
+        results.getEntries().forEach((key, value) -> {
+            log.info("Key table name: {}", key.getTableName());
+            String tableName = key.getTableName();
+            switch (tableName) {
+                case LR_REGISTRATION_TABLE_NAME:
+                    handleRegistrationTableEntries(value);
+                    break;
+                case LR_MODEL_METADATA_TABLE_NAME:
+                    handleClientMetadataTableEntries(value);
+                    break;
+                default:
+                    break;
+            }
+        });
+    }
+
+    /**
+     * Handle client registration table entries.
+     * @param registrationTableEntries list of client registration table entries
+     */
+    private void handleRegistrationTableEntries(List<CorfuStreamEntry> registrationTableEntries) {
+        if (registrationTableEntries == null) {
+            log.warn("No client registration table entries found!");
+            return;
+        }
+
+        for (CorfuStreamEntry entry : registrationTableEntries) {
+            if (entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
+                String clientName = ((ClientRegistrationId) entry.getKey()).getClientName();
+                ReplicationModel model = ((ClientRegistrationInfo) entry.getPayload()).getModel();
+                LogReplication.ReplicationSubscriber subscriber = LogReplication.ReplicationSubscriber.newBuilder()
+                        .setClientName(clientName).setModel(model).build();
+                // TODO (V2): Currently we add a default subscriber for logical group use case instead of listening
+                //  on client registration. Subscriber should be added upon registration after grpc stream for session
+                //  creation is added.
+//                configManager.onNewClientRegister(subscriber);
+//                sessionManager.createOutgoingSessionsBySubscriber(subscriber);
+                log.info("New client {} registered with model {}", clientName, model);
+            } else {
```

Comment:
can we add a log line for DELETE too that way we will know if the client ever uses that code path?